pyproject.toml

```diff
@@ -0,0 +1,155 @@
+[project]
+name = "snuba"
+version = "25.8.0.dev0"
+dependencies = [
+    "blinker>=1.5",
+    "click>=8.1.7",
+    "clickhouse-driver>=0.2.9",
+    "confluent-kafka>=2.7.0",
+    "datadog>=0.21.0",
+    "devservices>=1.2.1",
+    "fastjsonschema>=2.16.2",
+    "flask>=2.2.5",
+    "freezegun>=1.2.2",
+    "google-api-core>=2.19.1",
+    "google-api-python-client>=2.88.0",
+    "google-cloud-storage>=2.18.0",
+    "googleapis-common-protos>=1.63.2",
+    "jsonschema>=4.23.0",
+    "packaging>=24.1",
+    "parsimonious>=0.10.0",
+    "progressbar2>=4.2.0",
+    "proto-plus>=1.24.0",
+    "protobuf>=5.29.5",
+    "python-dateutil>=2.8.2",
+    "python-jose[cryptography]>=3.3.0",
+    "python-rapidjson>=1.8",
+    "pyuwsgi>=2.0.23",
+    "pyyaml>=6.0",
+    "redis>=4.5.4",
+    "sentry-arroyo>=2.29.1",
+    "sentry-kafka-schemas>=2.0.2",
+    "sentry-protos>=0.3.3",
+    "sentry-redis-tools>=0.5.0",
+    "sentry-relay>=0.9.5",
+    "sentry-sdk>=2.35.0",
+    "sentry-usage-accountant>=0.0.11",
+    "simplejson>=3.17.6",
+    "snuba-sdk>=3.0.39",
+    "sql-metadata>=2.11.0",
+    "sqlparse>=0.5.0",
+    "structlog>=22.3.0",
+    "structlog-sentry>=2.0.0",
+    "urllib3>=2.2.3",
+    "werkzeug>=3.0.6",
+    "rust_snuba",
+]
+
+[tool.uv.sources]
+rust_snuba = { workspace = true }
+
+[tool.uv.workspace]
+members = ["rust_snuba"]
+
+[tool.uv]
+environments = ["sys_platform == 'darwin' or sys_platform == 'linux'"]
+
+[[tool.uv.index]]
+# We don't allow using public pypi - submit a PR to
+# https://github.com/getsentry/pypi to add a dependency.
+url = "https://pypi.devinfra.sentry.io/simple"
+default = true
+
+[build-system]
+requires = ["uv_build==0.8.2"]
+build-backend = "uv_build"
+
+[tool.uv.build-backend]
+module-root = ""
+
+[project.scripts]
+snuba = "snuba.cli:main"
+
+[dependency-groups]
+dev = [
+    "black>=24.3.0",
+    "devservices>=1.2.1",
+    "flake8>=7.0.0",
+    "freezegun>=1.2.2",
+    "honcho>=1.1.0",
+    "isort==5.12.0",
+    "mypy>=1.1.1",
+    # pip required for maturin develop
+    # (can remove with maturin 1.6.0)
+    "pip>=23.3.2",
+    "pre-commit>=4.2.0",
+    "pytest>=8.3.3",
+    "pytest-cov>=4.1.0",
+    "pytest-watch>=4.2.0",
+    "sentry-devenv>=1.22.2",
+    "time-machine>=2.13.0",
+    "types-google-cloud-ndb>=2.2.0",
+    "types-protobuf>=5.27.0.20240626",
+    "types-python-dateutil>=2.8.19.14",
+    "types-python-jose>=3.3.0",
+    "types-pyyaml>=6.0.12.20240808",
+    "types-requests>=2.32.0.20240907",
+    "types-setuptools>=74.1.0.20240907",
+    "types-simplejson>=3.17.7",
+    "typing-extensions>=4.12.2",
+]
+
+[tool.black]
+# File filtering is taken care of in pre-commit.
+line-length = 100
+target-version = ['py311']
+
+[tool.isort]
+profile = "black"
+line_length = 100
+lines_between_sections = 1
+
+[tool.mypy]
```

Comment:
i also took the liberty of combining the previous mypy.ini and backend-typing flags into here in an improved more compact format!